lib/Twig/Template.php

```diff
@@ -305,6 +305,33 @@ public function getBlocks()
     }
 
     /**
+     * Returns the template source code.
+     *
+     * @return string|null The template source code or null if it can't be retrieved
```

Comment:
Why not throw an exception instead of returning null? I mean the null case is an error condition. And IMO multi-values return types should not be used for that.